src/OpenTelemetry/Metrics/BatchMetricPoint.cs

```diff
@@ -84,17 +84,22 @@ public void Dispose()
             /// <inheritdoc/>
             public bool MoveNext()
             {
-                var metricPoints = this.metricsPoints;
-                if (metricPoints[this.index].StartTime == default)
+                if (this.metricsPoints.Span[this.index].StartTime == default)
```

Comment:
In this method we access `this.metricsPoints.Span[this.index]` three times I think we could do something like...

```csharp
ref MetricPoint current = ref this.metricsPoints.Span[this.index];
```

...to capture a local ref to the underlying struct in the local scope.